rusty_pizza_server/src/order_model/meals.rs

```diff
@@ -31,11 +31,33 @@ impl Meals {
         self.meals.insert(id, meal);
         self.meals.get_mut(&id).unwrap()
     }
+
+    pub fn set_paid(&mut self, paid: Money) {
+        self.paid = paid;
+    }
+
+    pub fn set_tip(&mut self, tip: Money) {
+        self.tip = tip;
+    }
+
+    pub fn calculate_total_price(&self) -> Money {
+        let mut total_price = Money::new(0, 0);
+        for (_, meal) in self.meals.iter() {
+            total_price = total_price + meal.get_price();
+        }
+        return total_price;
```

Comment:
The whole function could be written as
```rust
pub fn calculate_total_price(&self) -> Money {
    self.meals.values().sum()
}
```
given that we implement [`std::iter::Sum`](https://doc.rust-lang.org/std/iter/trait.Sum.html) for `Money`.

Reply:
I don´t think it would by nice to shorten the method like this. It looks like sum up all `Meal`s and at the end we only sum up the prices

Reply:
You are right, I missed that the prices need to be extracted first:
```rust
pub fn calculate_total_price(&self) -> Money {
    self.meals.values().map(|meal| meal.get_price()).sum()
}
```

Reply:
I don´t think this is prettier than a for loop